crates/doc/src/ptr.rs

```diff
@@ -196,6 +203,7 @@ impl Pointer {
                     Token::Index(ind) => map.entry(ind.to_string()).or_insert(Value::Null),
                     Token::Property(prop) => map.entry(prop).or_insert(Value::Null),
                     Token::NextIndex => map.entry("-").or_insert(Value::Null),
```

Comment:
both `NextIndex` and `NextProperty` should return None here. They can't be created as properties.

(Technically `NextIndex` currently needs to create a "-" property, but won't once we no longer parse "-" into NextIndex).